src/controllers/winkels.ts

```diff
@@ -7,7 +7,26 @@ import { Store } from "../models/store";
  * Shows all stores
  */
 export async function getStores(req: Request, res: Response): Promise<void> {
-  const stores = await getManager().find(Store);
+  const manager = await getManager();
+  let stores;
+
+  if (Object.keys(req.query).length !== 0) {
+    // A search query was passed
+    let query = manager.createQueryBuilder(Store, "store");
+    if (req.query.name_desc !== undefined) {
+      query = query
+        .where("store.name like :nameDesc", { nameDesc: "%" + req.query.name_desc + "%" })
```

Comment:
If we want to keep using the ActiveRecord approach, you can use `Like`:

```typescript
import {Like} from "typeorm";

Store.find({ where: { title: Like(req.query.name_desc), description: Like(req.query.name_desc) })
```

It should be possible to construct the query further like this. Maybe you will have to use `find` instead of `where`. The [docs](https://github.com/typeorm/typeorm/blob/master/docs/find-options.md) on their GitHub seems to be better than their site...

Reply:
The query is built in 2 steps here, because if we have both a postal code and a name/description query, I want to query like this: `SELECT ... WHERE (store.name like name_desc OR store.description like name_desc) AND WHERE postcode = postal` (this might not be valid SQL but you get the point).

The docs describe how OR can be implemented by passing an array to `where`:
```
Store.find({ where: [{name: Like("%" + req.query.name_desc + "%")}, {description: Like("%" + req.query.name_desc + "%")}])
```
But I don't find how I can add an AND clause on top of this, without nesting it within the OR clause. If I do nest it within the OR clause, the code becomes less readable in my opinion:
```
  if (Object.keys(req.query).length !== 0) {
    // A search query was passed
    let queryDict = {where: [] as Record<string, FindOperator<any>>[]};
    if (req.query.name_desc !== undefined) {
      queryDict.where.push({name: Like("%" + req.query.name_desc + "%")});
      queryDict.where.push({description: Like("%" + req.query.name_desc + "%")});
      if (req.query.postal !== undefined) {
        queryDict.where.forEach(obj => {
          obj.postcode = Like(req.query.postal)
        });
      }
    } else if (req.query.postal !== undefined) {
      queryDict.where.push({postcode: req.query.postal});
    }
    stores = await Store.find(queryDict);
  } else {
    stores = await Store.find();
  }
```

Reply:
ActiveRecord approach is nicer in simpler cases though, but when nesting queries I think using a builder makes more sense